### DISCUSSION TITLE
How to reset list-item counter for custom lists?

### DISCUSSION THREAD

**Discussion by sametweb** [tailwindlabs/tailwindcss#5942] [Help]
Hello everyone,

I am facing a problem when rendering some nested markdown titles as `list-item`.

This is my renderer:
```jsx
const ContentMarkdownRenderer = () => {
  return {
    h1(props: any) {
      return (
	<h1 className='list-item list-inside list-decimal text-base pt-1 mb-1 pl-0'>
	  {props.children}
	</h1>
      )
    },
    h2(props: any) {
      return (
        <h2 className='list-item list-inside list-alpha text-sm pt-1 mb-1 pl-4'>
	  {props.children}
        </h2>
      )
    },
    h3(props: any) {
      return (
        <h3 className='list-item list-inside list-disc text-sm pt-1 mb-1 pl-8'>{props.children}</h3>
      )
    },
    h4(props: any) {
      return (
        <h4 className='list-item list-inside list-none text-sm pt-1 mb-1 pl-12'>
	  {props.children}
        </h4>
      )
    },
}
}
```

This code basically renders h1, h2, h3, and h4 elements from my markdown text. I am using this code to render some table of contents. My problem is, this code renders the titles and their `list-style-type` but they are not reset after a nested block comes in. In the picture, after `6`, nested block starts with `g` instead of `a`.

![image](https://user-images.githubusercontent.com/3162574/139579237-a249b8b3-a4ab-436b-be91-05eb7808ec2d.png)

I researched but couldn't find a solution. How should I reset the counter of the `list-style-type` in Tailwind when new type of element comes in? CSS has `counter-increment` and `counter-reset` properties but I am not sure how to implement them in my code. Any help is appreciated.

**Comment by Taiterbase** [comment]
👋 Tailwindcss has the `[ ]` syntax for when you need inline css [(src)](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties). You can use something like `[counter-reset: list-number]` in your `className` string. [This guide](https://blog.logrocket.com/styling-numbered-lists-with-css-counters/) has a pretty good walk-through that might help with the styling portion.
